pref/internal/battery/battery_js.go

```diff
@@ -0,0 +1,83 @@
+package battery
+
+import (
+	"errors"
+	"math"
+	"syscall/js"
+)
+
+var (
+	_GetBattery = js.Global().Get("navigator").Get("getBattery")
+)
+
+func batteryLevel() uint8 {
```

Comment:
If the device doesn't use a battery, it's always fully charged, so it returns 100.

However, I include a new error, which may happen when batteries don't power the device. Some API doesn't provide such information, so it will return 100 without error. The JS API returns "100" when the machine doesn't use a battery, and there's no way to distinguish that (not officially).